src/kr/kr1/cyclic.c

```diff
@@ -0,0 +1,17 @@
+#include <stdbool.h>
+int find_max(bool* bits, int n)
+{
+    int max = 0;
+    for (int s = 0; s < n; s++) {
+        int num = 0;
+        for (int i = 0; i < n; i++) {
+            if (bits[(s + i) % n])
+                num = (num << 1) | 1;
+            else
+                num = num << 1;
```

Comment:
Кажется, можно сделать так:

```suggestion
            num <<= 1;
            num |= bits[(s + i) % n];
```
На мой взгляд, так лучше понятно, что тут делается: число сдвигается влево, а самый левый бит переезжает вправо.